python/simulation.py

```diff
@@ -34,6 +34,20 @@ def get_num_args(func):
     return func.__code__.co_argcount
 
 
+def vec(*args):
+    try:
+        # Check for vec(iterable)
+        return mp._vec(*args[0])
```

Comment:
This will silently ignore other arguments if the first argument is iterable.

Shouldn't it try `mp._vec(*args[0])` if `arg.len == 1` and `mp._vec(*args)` otherwise?

Reply:
Updated to handle all malformed arguments.